cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkRuntimeService.java

```diff
@@ -225,9 +227,14 @@ public SparkExecutionContext create(SparkRuntimeContext runtimeContext) {
 
         // Localize the hConf file to executor nodes
         localizeResources.add(new LocalizeResource(saveHConf(hConf, tempDir)));
+
+        extraJars = CConfigurationUtil.getExtraJarNames(cConf);
```

Comment:
This is not good. We should only have one `List<File> getExtraJars` method and have the `CConfiguration` updated when launching a container (in `AbstractDistributedProgramRunner.saveCConf` method).